### PR TITLE
#17478 remove page from velocityCache if template is unpublished

### PR DESCRIPTION
**Pull request by erickgonzalez**

### CODE REVIEW

**Comment by wezell** [comment]
Erick, we can't do this logic as there can be thousands of pages on a site that uses a template.  We need to deal with this case in the Velocity and or PageLoader

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java [hunk 1]
```diff
@@ -126,6 +129,13 @@ public final void serve(final OutputStream out) throws DotDataException, IOExcep
                 }
             }
 
+            //Validate if template is publish, if not remove page from cache
+            if (!UtilMethods.isSet(APILocator.getTemplateAPI().findLiveTemplate(htmlPage.getTemplateId(),APILocator.systemUser(),false).getInode())) {
```
Comment (wezell) [inline dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: All our front end (delivery tier) calls need to use cache.  Please insure that this call is cached and also that the templateAPI has a 404 cache.  If it doesn't, you might want to just call the `VersionableAPI.findLiveVersion` which does have a 404 cache.

**Review by wezell — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java (hunk 1 above)
Comment (erickgonzalez) [reply dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityLiveMode.java]: Looking at the TemplateAPI and internally calls the versionableFactory that contains 404 cache as you mentioned.

**Review by wezell — APPROVED** [review]
looks good

**Review by dsilvam — APPROVED** [review]

**Pull request merged**